autoload/rust.vim

```diff
@@ -508,16 +508,23 @@ function! s:SearchTestFunctionNameUnderCursor() abort
     return matchstr(getline(test_func_line), '\m\C^\s*fn\s\+\zs\h\w*')
 endfunction
 
-function! rust#Test(all, options) abort
+function! rust#Test(mods, winsize, all, options) abort
     let manifest = findfile('Cargo.toml', expand('%:p:h') . ';')
     if manifest ==# ''
         return rust#Run(1, '--test ' . a:options)
     endif
 
+    " <count> defaults to 0, but we prefer an empty string
+    let winsize = a:winsize ? a:winsize : ''
+
     if has('terminal')
-        let cmd = 'terminal '
+        if has('patch-8.0.910')
+            let cmd = printf('%s noautocmd %snew | terminal ++curwin ', a:mods, winsize)
```

Comment:
I prefer `printf()`, but if you want, I change it to the even more self-explaining:

```vim
let cmd = a:mods . ' noautocmd ' . winsize . 'new | terminal ++curwin'
```